integration_tests/src/main/python/array_test.py

```diff
@@ -625,3 +625,35 @@ def test_arrays_overlap_before_spark313(data_gen):
             'arrays_overlap(array(3, 4), array(1, 2))',
             'arrays_overlap(array(), array(1, 2))')
     )
+
+@pytest.mark.parametrize('data_gen', [byte_gen, short_gen, int_gen, long_gen], ids=idfn)
```

Comment:
Would be nice here to increase the probability that 0, -10 and 10 show up in the data gen, so it is much more likely, especially for long_gen and int_gen that they will show up.